model/textparse/openmetricsparse_test.go

```diff
@@ -119,356 +119,545 @@ foobar{quantile="0.99"} 150.1`
+		t.Run(fmt.Sprintf("type-and-unit=%v", typeAndUnitEnabled), func(t *testing.T) {
+			exp := []parsedEntry{
+				{
+					m:    "go_gc_duration_seconds",
+					help: "A summary of the GC invocation durations.",
+				}, {
+					m:   "go_gc_duration_seconds",
+					typ: model.MetricTypeSummary,
+				}, {
+					m:    "go_gc_duration_seconds",
+					unit: "seconds",
+				}, {
+					m: `go_gc_duration_seconds{quantile="0"}`,
+					v: 4.9351e-05,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "__type__", string(model.MetricTypeSummary), "__unit__", "seconds", "quantile", "0.0"),
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "quantile", "0.0"),
+					),
+				}, {
+					m: `go_gc_duration_seconds{quantile="0.25"}`,
+					v: 7.424100000000001e-05,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "__type__", string(model.MetricTypeSummary), "__unit__", "seconds", "quantile", "0.25"),
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "quantile", "0.25"),
+					),
+				}, {
+					m: `go_gc_duration_seconds{quantile="0.5",a="b"}`,
+					v: 8.3835e-05,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "__type__", string(model.MetricTypeSummary), "__unit__", "seconds", "quantile", "0.5", "a", "b"),
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "quantile", "0.5", "a", "b"),
+					),
+				}, {
+					m:    "nohelp1",
+					help: "",
+				}, {
+					m:    "help2",
+					help: "escape \\ \n \\ \" \\x chars",
+				}, {
+					m:    "nounit",
+					unit: "",
+				}, {
+					m: `go_gc_duration_seconds{quantile="1.0",a="b"}`,
+					v: 8.3835e-05,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "__type__", string(model.MetricTypeSummary), "quantile", "1.0", "a", "b"),
+						labels.FromStrings("__name__", "go_gc_duration_seconds", "quantile", "1.0", "a", "b"),
+					),
+				}, {
+					m: `go_gc_duration_seconds_count`,
+					v: 99,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						labels.FromStrings("__name__", "go_gc_duration_seconds_count", "__type__", string(model.MetricTypeSummary)),
+						labels.FromStrings("__name__", "go_gc_duration_seconds_count"),
+					),
+				}, {
+					m: `some:aggregate:rate5m{a_b="c"}`,
+					v: 1,
+					lset: typeAndUnitLabels(
+						typeAndUnitEnabled,
+						// NOTE(bwplotka): This is knowingly broken, inheriting old type when TYPE was not specified on a new metric.
+						// This was broken forever on a case for a broken exposition. Don't fix for now (expensive).
+						labels.FromStrings("__name__", "some:aggregate:rate5m", "__type__", string(model.MetricTypeSummary), "a_b", "c"),
+						labels.FromStrings("__name__", "some:aggregate:rate5m", "a_b", "c"),
```

Comment:
Ah, I was finally able to understand the bug you mentioned after reading this test.

This sounds like a very serious bug actually. I know the bug will be hidden behind a feature flag, so it's fine to merge, but maybe we want to document this behavior in `docs/feature_flags.md`?

Reply:
Yea, I think it's ok to NOT put in the docs - this feature is very much in dev, there are more bugs and gaps e.g. OTLP and PRW support, so maybe creating meta issue would be more appriopriate and linking that?

Reply:
Done: https://github.com/prometheus/prometheus/issues/16610